src/main.c

```diff
@@ -463,42 +459,46 @@ GLOBAL_ASM("asm/non_matchings/main/dma_copy.s")
 // Resembles setup_game_memory from SM64
 #ifdef NON_MATCHING
 void init_game(void) {
-    s32 sp40;
-    s32 sp38;
+    UNUSED s32 sp38;//0x38(sp)
     u32 *sp2C;
-    s32 temp_a1;
     s32 temp_t7;
+    s32 sp40;
     s32 temp_t2;
+    u32 temp_a1;
+	UNUSED s16 unknown_padding;
 
     func_800010CC();
-    gHeapEndPtr = 0x8028DF00;
+    gHeapEndPtr = SEG_8028DF00;
     set_segment_base_addr(0, 0x80000000);
     func_802A7CF0(&D_801978D0, 0x80242F00);
     func_80000BEC();
-    osInvalDCache(0x802BA370, 0x5810);
-    osPiStartDma(&D_8014F0A0, 0, 0, &_data_802BA370SegmentRomStart, 0x802BA370, 0x5810, &D_8014EF58);
+    osInvalDCache(SEG_802BA370, 0x5810);
+    osPiStartDma(&D_8014F0A0, 0, 0, &_data_802BA370SegmentRomStart, SEG_802BA370, 0x5810, &D_8014EF58);
     osRecvMesg(&D_8014EF58, &D_8014F098, 1);
     set_segment_base_addr(2, func_802A7D70(&_data_segment2SegmentRomStart, &_data_segment2SegmentRomEnd));
     temp_t2 = ALIGN16((u32)&_common_texturesSegmentRomEnd - (u32)&_common_texturesSegmentRomStart);
-    sp2C = 0x8028DF00 - temp_t2;
-    osPiStartDma(&D_8014F0A0, 0, 0, &_common_texturesSegmentRomStart, 0x8028DF00 - temp_t2, temp_t2, &D_8014EF58);
+    sp2C = SEG_8028DF00 - temp_t2;
+    osPiStartDma(&D_8014F0A0, 0, 0, &_common_texturesSegmentRomStart, SEG_8028DF00 - temp_t2, temp_t2, &D_8014EF58);
     osRecvMesg(&D_8014EF58, &D_8014F098, 1);
+	
+	
+	//need to match this
+	
     temp_a1 = gPrevLoadedAddress;
-    sp38 = temp_a1;
     sp40 = ALIGN16(sp2C[1]);
     mio0decode(sp2C, temp_a1);
     set_segment_base_addr(0xD, temp_a1);
-    temp_t7 = sp40 + gPrevLoadedAddress;
```

Comment:
I think the issue with matching here is likely temp_t7. It probably needs to disappear and instead it adds back and forth between either sp40 or gPrevLoadedAddress.
Ex:
```
gPrevLoadedAddress + sp40;
```